indico_toolkit/types/predictions.py

```diff
@@ -1,134 +1,22 @@
-
-    def remove_keys(self, keys_to_remove: List[str] = ["start", "end"]):
-        for pred in self._preds:
-            for key in keys_to_remove:
-                pred.pop(key)
-
-    def remove_human_added_predictions(self):
-        """
-        Remove predictions that were not added by the model (i.e. added by scripted or human review)
-        """
-        self._preds = [
-            i for i in self._preds if not self.is_manually_added_prediction(i)
-        ]
-
-    def to_list(self):
-        return self._preds
-
     @staticmethod
-    def is_manually_added_prediction(prediction: dict) -> bool:
-        if isinstance(prediction["start"], int) and isinstance(prediction["end"], int):
-            if prediction["end"] > prediction["start"]:
-                return False
-        return True
-
-    @staticmethod
-    def get_extraction_labels_set(predictions: List[dict]) -> Set[str]:
-        return set([i["label"] for i in predictions])
-
-    def _select_max_confidence(self, label):
-        max_pred = None
-        confidence = 0
-        for pred in self[label]:
-            pred_confidence = pred["confidence"][label]
-            if pred_confidence >= confidence:
-                max_pred = pred
-                confidence = pred_confidence
-        return max_pred
-
-    @property
-    def label_count_dict(self) -> Dict[str, int]:
-        """
-        Get count of occurrences of each label
-        """
-        return dict(Counter(i["label"] for i in self._preds))
-
-    def _remove_all_by_label(self, label):
-        for pred in self._preds:
-            if pred["label"] == label:
-                self._preds.remove(pred)
-
-    def __len__(self):
-        return len(self._preds)
-
-    def __repr__(self):
-        return f"Prediction Class, {self.num_predictions} Predictions:\n\n{self._preds}"
-
-    def __getitem__(self, label: str) -> List[dict]:
-        return [i for i in self._preds if i["label"] == label]
-
-    def to_csv(
-        self, save_path: str, filename: str = "", append_if_exists: bool = True
-    ) -> None:
+    def get_obj(predictions):
         """
-        Write three column CSV ('confidence', 'label', 'text')
-        Args:
-            save_path (str): path to write CSV 
-            filename (str, optional): the file where the preds were derived from. Defaults to "".
+        Returns:
+        Extractions object or Classifications object depending on predictions type
         """
-        self.set_confidence_key_to_max_value()
-        self.remove_keys(keys_to_remove=["start", "end"])
-        df = pd.DataFrame(self._preds)
-        df["filename"] = filename
-        if append_if_exists and FileProcessing.file_exists(save_path):
-            df.to_csv(save_path, mode="a", header=False, index=False)
+        if type(predictions) == list:
+            return indico_toolkit.types.Extractions(predictions)
+        elif type(predictions) == dict:
```

Comment:
I tend to prefer 'isinstance(predictions, dict)', but no strong feeling if you want to keep as is
https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance